### ISSUE TITLE
Straight and Flowchart Connectors drawn in reverse of Bezier and State Machine Connectors

### ISSUE THREAD

**Issue by dbnoble** [sporritt/jsPlumb#476]
It seems that the Straight and Flowchart connections draw the path from `endpoint[1]` to `endpoint[0]`, while Bezier and State Machine connectors draw from `endpoint[0]` to `endpoint[1]`, or maybe vice-versa.

Example: http://codepen.io/vertical3/pen/ZQxwqd

**Comment by sporritt** [comment]
i dont think so..if you set an overlay at position 1, it always ends up at the target element, regardless of the connector tye.

**Comment by sporritt** [comment]
..oh you mean because of this animation stuff you're doing.  maybe it's true.  i don't consider this a bug though.

**Comment by dbnoble** [comment]
The animation isn't completely relevant, it's more for showing the effect in this case, but it would cause problems if you were wanting to animate using dash-strokeoffset or an outside library like Greensock's drawSVG, or someone who was trying to do layered dash-arrays I imagine. It could potentially cause other issues that I'm not aware of.

If the path is indeed getting drawn from the opposite origin, it's not a bug so much as an inconsistency. Is there perhaps some other reason that might be happening?

**Comment by sporritt** [comment]
ah yes well moving forward with sprout means fixing these things.
